### PR TITLE
Some stubs for better compatibility

### PR DESCRIPTION
**Pull request by dima-xd**
Not sure about `GetCurrentIpAddress`, so please inform if I'm wrong here

### CODE REVIEW

File: app/src/main/cpp/skyline/applet/error_applet.h [hunk 1]
```diff
@@ -41,6 +41,12 @@ namespace skyline::applet {
         };
         static_assert(sizeof(ErrorCommonHeader) == 0x8);
 
+        struct ErrorCommonArg {
```
Comment (bylaws) [inline app/src/main/cpp/skyline/applet/error_applet.h:44]: Add link to switchbrew

File: app/src/main/cpp/skyline/services/account/IAccountServiceForApplication.cpp [hunk 2]
```diff
@@ -73,6 +73,20 @@ namespace skyline::service::account {
...
 
+    Result IAccountServiceForApplication::IsUserRegistrationRequestPermitted(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        response.Push<u8>(false); // Registration isn't permitted via the application account service
+        return {};
+    }
+
+    Result IAccountServiceForApplication::TrySelectUserWithoutInteraction(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto isNetworkServiceAccountRequired{request.Pop<u8>()};
+        if (isNetworkServiceAccountRequired)
+            Logger::Info("Network service account is required for user selection");
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/account/IAccountServiceForApplication.cpp:84]: Do we really need a log here?

File: app/src/main/cpp/skyline/services/am/controller/IApplicationFunctions.h [hunk 3]
```diff
@@ -79,6 +79,12 @@ namespace skyline::service::am {
          */
         Result SetGamePlayRecordingState(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
 
+        /**
+         * @brief Enables application crash report
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/am/controller/IApplicationFunctions.h:83]: can drop this it's obvious

**Review by bylaws — CHANGES_REQUESTED** [review]

File: app/src/main/cpp/skyline/applet/web_applet.cpp [hunk 4]
```diff
@@ -21,7 +21,9 @@ namespace skyline::applet {
         auto commonArg{PopNormalInput<service::applet::CommonArguments>()};
         auto argHeader{PopNormalInput<WebArgHeader>()};
 
-        if ((commonArg.apiVersion >= 0x80000 && argHeader.shimKind == ShimKind::Web) || (commonArg.apiVersion >= 0x30000 && argHeader.shimKind == ShimKind::Share))
+        if ((commonArg.apiVersion >= 0x80000 && argHeader.shimKind == ShimKind::Web) ||
+            (commonArg.apiVersion >= 0x50000 && argHeader.shimKind == ShimKind::Shop) ||
```
Comment (bylaws) [inline app/src/main/cpp/skyline/applet/web_applet.cpp:25]: Shop doesn't use tlvs so you can leave this line as before

File: app/src/main/cpp/skyline/services/visrv/IApplicationDisplayService.cpp [hunk 5]
```diff
@@ -40,6 +40,21 @@ namespace skyline::service::visrv {
         return {};
     }
 
+    Result IApplicationDisplayService::ListDisplays(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        struct DisplayInfo {
+            char displayName[0x40]{"Default"};
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/visrv/IApplicationDisplayService.cpp:45]: Std array<u8, 0x40>

File: app/src/main/cpp/skyline/services/visrv/IApplicationDisplayService.cpp [hunk 6]
```diff
@@ -40,6 +40,21 @@ namespace skyline::service::visrv {
...
+            u8 hasLimitedLayers{1};
+            INSERT_PADDING_BYTES(7);
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/visrv/IApplicationDisplayService.cpp:47]: U8 _pad_[7]

**Review by bylaws — CHANGES_REQUESTED** [review]

File: app/src/main/cpp/skyline/services/account/IProfile.h [hunk 7]
```diff
@@ -18,6 +18,18 @@ namespace skyline::service::account {
       private:
         UserId userId;
 
+        static constexpr std::array<u8, 107> profileImageIcon{
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/account/IProfile.h:21]: What is this

Comment (bylaws) [reply app/src/main/cpp/skyline/services/account/IProfile.h:21]: I would also just store it in the cpp file itself

Comment (dima-xd) [reply app/src/main/cpp/skyline/services/account/IProfile.h:21]: > What is this

image, in bytes array

Comment (bylaws) [reply app/src/main/cpp/skyline/services/account/IProfile.h:21]: I know that lol but what's the image

Comment (dima-xd) [reply app/src/main/cpp/skyline/services/account/IProfile.h:21]: > I know that lol but what's the image

the smallest image I could find https://github.com/mathiasbynens/small/blob/master/jpeg.jpg

Comment (bylaws) [reply app/src/main/cpp/skyline/services/account/IProfile.h:21]: Ah, move this to the cpp file and comment as such then

File: app/src/main/cpp/skyline/applet/error_applet.h [hunk 8]
```diff
@@ -54,6 +63,8 @@ namespace skyline::applet {
 
         std::shared_ptr<service::am::IStorage> errorStorage;
 
+        void HadndleErrorCommonArg();
```
Comment (lynxnb) [inline app/src/main/cpp/skyline/applet/error_applet.h:66]:
```suggestion
        void HandleErrorCommonArg();
```

Comment (dima-xd) [reply app/src/main/cpp/skyline/applet/error_applet.h]: Thanks for noticing

**Pull request merged**